crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,235 @@
+            encoded_key,
+            Some(password.ok_or(SshKeyImportError::PasswordRequired)?),
+        ),
+        OPENSSH_LABEL => import_openssh_key(encoded_key, password),
+        _ => Err(SshKeyImportError::ParsingError),
+    }
+}
+
+fn import_pkcs8_key(
+    encoded_key: String,
+    password: Option<String>,
+) -> Result<SshKey, SshKeyImportError> {
+    let der = if let Some(password) = password {
+        SecretDocument::from_pkcs8_encrypted_pem(&encoded_key, password.as_bytes()).map_err(
+            |err| match err {
+                pkcs8::Error::EncryptedPrivateKey(pkcs5::Error::DecryptFailed) => {
+                    SshKeyImportError::WrongPassword
+                }
+                _ => SshKeyImportError::ParsingError,
+            },
+        )?
+    } else {
+        SecretDocument::from_pkcs8_pem(&encoded_key).map_err(|_| SshKeyImportError::ParsingError)?
+    };
+
+    let private_key_info =
+        PrivateKeyInfo::from_der(der.as_bytes()).map_err(|_| SshKeyImportError::ParsingError)?;
+
+    let private_key = match private_key_info.algorithm.oid {
+        ed25519::pkcs8::ALGORITHM_OID => {
+            let private_key: ed25519::KeypairBytes = private_key_info
+                .try_into()
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+
+            ssh_key::private::PrivateKey::from(Ed25519Keypair::from(&private_key.secret_key.into()))
+        }
+        rsa::pkcs1::ALGORITHM_OID => {
+            let private_key: rsa::RsaPrivateKey = private_key_info
+                .try_into()
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+
+            ssh_key::private::PrivateKey::from(
+                RsaKeypair::try_from(private_key).map_err(|_| SshKeyImportError::ParsingError)?,
+            )
+        }
+        _ => return Err(SshKeyImportError::UnsupportedKeyType),
+    };
+
+    private_key
+        .try_into()
+        .map_err(|_| SshKeyImportError::ParsingError)
+}
+
+fn import_openssh_key(
+    encoded_key: String,
+    password: Option<String>,
+) -> Result<SshKey, SshKeyImportError> {
+    let private_key =
+        ssh_key::private::PrivateKey::from_openssh(&encoded_key).map_err(|err| match err {
+            ssh_key::Error::AlgorithmUnknown | ssh_key::Error::AlgorithmUnsupported { .. } => {
+                SshKeyImportError::UnsupportedKeyType
+            }
+            _ => SshKeyImportError::ParsingError,
+        })?;
+
+    if private_key.is_encrypted() {
+        if let Some(password) = password {
+            private_key
+                .decrypt(password.as_bytes())
+                .map_err(|_| SshKeyImportError::WrongPassword)?
+                .try_into()
+                .map_err(|_| SshKeyImportError::ParsingError)
+        } else {
+            Err(SshKeyImportError::PasswordRequired)
+        }
+    } else {
+        private_key
+            .try_into()
+            .map_err(|_| SshKeyImportError::ParsingError)
+    }
```

Comment:
**suggestion**: We can improve this code slightly.

- The if condition for password can be rewritten using `ok_or` which converts an `Option` to `Result`.
- We can remove some code duplication by assigning a temporary variable and lifting out some code from the if block.

```suggestion
    let private_key = if private_key.is_encrypted() {
        let password = password.ok_or(SshKeyImportError::PasswordRequired)?;
        private_key
            .decrypt(password.as_bytes())
            .map_err(|_| SshKeyImportError::WrongPassword)?
    } else {
        private_key
    }

    private_key
        .try_into()
        .map_err(|_| SshKeyImportError::ParsingError)
```